experimental/overhead_matching/swag/evaluation/evaluate_swag.py

```diff
@@ -121,19 +121,28 @@ def run_inference_on_path(
     patch_similarity_for_path: torch.Tensor,  # path_length x W
```

Comment:
Consider making the changes to dump the filter state a separate PR and merging. It seems useful, even without the visualization part being ready yet.